.changesets/add-config-appsignal-rb-file-support.md

```diff
@@ -0,0 +1,19 @@
+---
+bump: minor
+type: add
+---
+
+Add `config/appsignal.rb` config file support. When a `config/appsignal.rb` file is present in the app, the Ruby gem will automatically load it when `Appsignal.start` is called.
+
+The `config/appsignal.rb` config file is a replacement for the `config/appsignal.yml` config file. When both files are present, only the `config/appsignal.rb` config file is loaded.
+
+Example `config/appsignal.rb` config file:
+
+```ruby
+# config/appsignal.rb
+Appsignal.configure do |config|
+  config.name = "My app name"
+end
+```
+
+It is not possible to configure multiple environments in the `config/appsignal.rb` config file as it was possible in the YAML version.
```

Comment:
Is it not? I thought you could do `Appsignal.configure(:test) do ... end`.

Reply:
No, it is not.

`Appsignal.configure(:test)` will configure `Appsignal.config` with env "test".
Then calling `Appsignal.configure(:production)` will configure `Appsignal.config` with env "production", overwriting any previously set config for the "test" env.

Reply:
For example, this will not work to define multiple environments. The last `Appsignal.configure` call is leading.

```ruby
Appsignal.configure(:test) do |config|
  config.ignore_actions << "My test action"
end

Appsignal.configure(:production) do |config|
  config.ignore_actions << "My production action"
end

# Appsignal.config[:ignore_actions] == ["My production action"]
```

In the end I just want to move to this situation where the config is mutable across envs, because it doesn't have to create a new config when a new env is set. But we can only reasonably do that when we don't have to consider the YAML file.

```ruby
Appsignal.configure do |config|
  config.env = "test"
  config.ignore_actions << "My test action"
end

Appsignal.configure do |config|
  config.env = "production"
  config.ignore_actions << "My production action"
end

# Appsignal.config[:ignore_actions] == ["My test action", "My production action"]
```

Reply:
So is there no way to have conditional configuration per environment on `config/appsignal.rb`? Something like:

```ruby
Appsignal.configure do |config|
  config.ignore_actions << "My production action" if config.env == "production"
end
```

Reply:
Yes that's possible, but you can't call `Appsignal.configure` with different envs multiple times.

Reply:
~~~suggestion
To configure different option values for environments in the `config/appsignal.rb` config file, use if-statements:

```ruby
# config/appsignal.rb
Appsignal.configure do |config|
  config.name = "My app name"

  if config.env == "production"
    config.ignore_actions << "My production action"
  end

  if config.env == "staging"
    config.ignore_actions << "My staging action"
  end
end
```
~~~